lib/posthog/client.ex

```diff
@@ -1,28 +1,54 @@
 defmodule Posthog.Client do
   @moduledoc false
 
+  defp headers(), do: headers(nil)
+  defp headers(nil), do: [{"Content-Type", "application/json"}]
+  defp headers(additional_headers) do
+    Enum.reduce(
+      additional_headers,
+      headers(nil),
+      fn header, headers ->
+        [header | headers]
+      end
+    )
+  end
+
+  def capture(event, params, opts) when is_list(opts), do: capture(event, params, Map.new(opts))
+  def capture(event, params, opts) when is_map(opts) do
+    headers = Map.get(opts, :headers) |> headers()
```

Comment:
you generally don't to use a map for headers because it doesn't allow for repetition of a single header name which is allowed in http